src/passes/PostEmscripten.cpp

```diff
@@ -75,6 +78,24 @@ struct OptimizeCalls : public WalkerPass<PostWalker<OptimizeCalls>> {
 
 struct PostEmscripten : public Pass {
   void run(PassRunner* runner, Module* module) override {
+    // Apply the stack pointer, if it was provided.  This is needed here
+    // bacuase emscripten JS compiler can add static data allocations that
+    // comes before the stack.
+    auto stackPtrStr =
+      runner->options.getArgumentOrDefault("stack-pointer", "");
+    if (stackPtrStr != "") {
+      Global* stackPointer = getStackPointerGlobal(*module);
+      BYN_TRACE("stack_pointer: " << stackPtrStr << "\n");
+      if (stackPointer && !stackPointer->imported()) {
```

Comment:
should we warn if we can't find a stack pointer? i'm not sure if that's an error case or not (do we not emit any stack in some wasm files, if they never need one?)

Reply:
I guess maybe there could be a program that doesn't use the stack pointer at all?

Reply:
I'm not sure we currently allow that though.